src/modules/fancyzones/lib/WindowMoveHandler.cpp

```diff
@@ -0,0 +1,360 @@
+#include "pch.h"
+#include "WindowMoveHandler.h"
+
+#include <common/notifications.h>
+#include <common/notifications/fancyzones_notifications.h>
+#include <common/window_helpers.h>
+
+#include "lib/Settings.h"
+#include "lib/ZoneWindow.h"
+#include "lib/util.h"
+
+extern "C" IMAGE_DOS_HEADER __ImageBase;
+
+class WindowMoveHandlerPrivate
+{
+public:
+    WindowMoveHandlerPrivate(const winrt::com_ptr<IFancyZonesSettings>& settings) :
+        m_settings(settings)
+    {};
+
+    bool IsDragEnabled() const noexcept
+    {
+        return m_dragEnabled;
+    }
+
+    bool InMoveSize() const noexcept
+    {
+        return m_inMoveSize;
+    }
+
+    void MoveSizeStart(HWND window, HMONITOR monitor, POINT const& ptScreen, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept;
+    void MoveSizeUpdate(HMONITOR monitor, POINT const& ptScreen, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept;
+    void MoveSizeEnd(HWND window, POINT const& ptScreen, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept;
+
+    void MoveWindowIntoZoneByIndexSet(HWND window, HMONITOR monitor, const std::vector<int>& indexSet, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap) noexcept;
+    bool MoveWindowIntoZoneByDirection(HMONITOR monitor, HWND window, DWORD vkCode, bool cycle, const std::map<HMONITOR, winrt::com_ptr<IZoneWindow>>& zoneWindowMap);
+
+private:
+    void UpdateDragState(HWND window) noexcept;
+
+    bool IsCursorTypeIndicatingSizeEvent();

```

Comment:
This could be moved to utils

Reply:
I had doubt for this since it isn't used anywhere else. On the other hand it possibly may be useful later. @enricogior what do you think about moving it to utils?

Reply:
It's very specific of this class, it can stay here for now.

Reply:
Maybe mark it as static so we know it does not use any members?

Reply:
or even moved outside of the class!

Reply:
OK move it outside of the class.

Reply:
Done!